### ISSUE TITLE
Audio Mic Recording results in silence

### ISSUE THREAD

**Issue by Foxomat** [godotengine/godot#69755]
### Godot version

3.5.1

### System information

Windows 10, GLES3

### Issue description

I wanted to implement voice chat in my game. I took a look into the ```Audio Mic Record Demo``` and noticed it wasn't working as intended:  
- When recording audio and then clicking on "play", you hear nothing.
- When saving the recorded audio to a file (using ```save_to_wav()``` in this case) the resulting file is only silence

I made sure my mic is working: when I disable "muted" on the record audio bus, I can hear my voice in the demo, so the mic input actually gets to the game.

May be related to #33184 

### Steps to reproduce

Follow the [Recording with microphone](https://docs.godotengine.org/en/stable/tutorials/audio/recording_with_microphone.html) Tutorial or try the [Demo Project](https://github.com/godotengine/godot-demo-projects/tree/master/audio/mic_record)

### Minimal reproduction project

[Demo Project](https://github.com/godotengine/godot-demo-projects/tree/master/audio/mic_record)

**Comment by Calinou** [comment]
Do your Windows privacy settings allow apps to access the microphone?

**Comment by fire** [comment]
Mic recording for voip uses like triggering every fraction of a second is not expected to work with AudioStreamRecord.

The expected use case for AudioStreamRecord is you start it and record periods of a few seconds.

https://docs.godotengine.org/en/stable/classes/class_audioeffectcapture.html was written for the use in voice chat systems.

**Comment by Foxomat** [comment]
> Do your Windows privacy settings allow apps to access the microphone (**Allow apps to access the microphone** setting)? The microphone permission in the Windows settings affects both Win32 and UWP applications, even though it doesn't look like it at a glance.

Yes. When running the demo I can see in the windows settings that it uses my mic. And as I wrote, when unmuting the record bus, I can hear myself in the demo.

**Comment by Foxomat** [comment]
> Mic recording for voip uses like triggering every fraction of a second and returning a buffer is not expected to work with AudioStreamRecord.
> 
> The expected use case for AudioStreamRecord is you start it and record a lengthy period.
> 
> https://docs.godotengine.org/en/stable/classes/class_audioeffectcapture.html was written for the use in voice chat systems.
> 
> For 4.0, we wrote a speech plugin for voip and has a networked demo.
> 
> https://github.com/V-Sekai/godot-speech

Yes, for my demo I try using the [Godot 3 VOIP plugin](https://github.com/ikbencasdoei/godot-voip) now.  



> So recording works?

I don't know which part of it doesn't work. I've written above when I can hear myself and when I don't.

**Comment by starcin** [comment]
I want to add to this.

I am having trouble with recording sound via microphone using both 3.4.2 stable and 4.0 RC1.
While looking through related issues, I've seen very recent PRs (#69120 and #72898) and cloned the latest state of master branch on Feb 16.
I've spent the whole day compiling (this is my first time compiling the engine on my own) and trying to make the recording work. `(v4.0.rc.custom_build [953383328])`

Combining every suggestion in multiple issues, I made some little achievements:
I've disabled `playing` and `autoplay` properties of the recording player. And I run `play()` at `_ready`. And unmuted the bus with Record effect.
This way, when I start recording in game, I can clearly hear my echo via speakers AND since I am printing the peak volume to a label, I can see the value changes as a speak. So it clearly recognizes the microphone and gets the sound.

But when I print the data array of the recording (`AudioStreamWAV.data`), it is full of zeros and nothing else. If I record the audio, I can see that the length of the audio file is correct but it is only silence.

I am also adding my minimal project.
[VoiceTest.zip](https://github.com/godotengine/godot/files/10763820/VoiceTest.zip)

**Comment by sanchitgulati** [comment]
same issue, with Godot4rc4. I can hear the mic if I unmute the Record Bus but `record_effect.get_recording()` gives me silence. 

While exporting the wav file, the size reflects the duration of the recording.
Basically short recording gets me a small wav file and a longer recording gives me a larger wav file, but there is no data in those files. It's just silence.

I am using the [mic_record sample](https://github.com/godotengine/godot-demo-projects/tree/master/audio/mic_record) (updated to Godot4).
OS: Windows 11

**Comment by Cannon-S** [comment]
Still having this issue with Godot 4.01 stable. Windows 10.

I followed https://docs.godotengine.org/en/stable/tutorials/audio/recording_with_microphone.html exactly on a new project as well. I also made sure to enabled audio input in project settings. Please help :'(

**Comment by marcol1** [comment]
yep, unfortunately for me uninstalling drivers didn't work (but I do have multiple). my laptop comes with Nahimic and other drivers, which is a pain to uninstall. hopefully this gets fixed soon!

**Comment by Triggered41** [comment]
@dev-mico 
Hi, I was fiddeling with AudioStreamRecord today and noticed that the Recording is empty (results in all 0's data) although the microphone is working in godot, I downloaded the official recording demo and the recording worked in the demo after a while I found out that

The audio records when
- An **AudioStreamPlayer** with stream as **AudioStreamMicrophone** has **autoplay** turned on and Bus set to "BusWithRecordEffect,"

if any of the above setting is different(like autoplay=false) result is silent.

Is it a bug or is it supposed to be this way, why?

Godot Ver: 4.6.1.stable.official

**Comment by LingLambda** [comment]
Confirmed on Windows 11 25H2 + Godot 4.6 + [SteelSeries Sonar](https://steelseries.com/gg/sonar#sonar), killing Sonar process fixes it.

I am using the [mic_record sample](https://github.com/godotengine/godot-demo-projects/tree/master/audio/mic_record) too.

I believe Sonar is somehow interfering with Godot's audio input, though I'm not sure of the exact cause. Even when I set the microphone directly as the default audio input device while Sonar is running — bypassing Sonar entirely — the problem still persists. Only after closing Sonar and then setting the microphone as default does Godot record properly.
